pkgs/tools/networking/i2pd/default.nix

```diff
@@ -1,19 +1,19 @@
-{ stdenv, fetchFromGitHub, fetchpatch, boost, zlib, openssl }:
+{ stdenv, fetchFromGitHub, fetchpatch, boost165, zlib, openssl }:
 
 stdenv.mkDerivation rec {
 
   name = pname + "-" + version;
   pname = "i2pd";
-  version = "2.15.0";
+  version = "2.17.0";
 
   src = fetchFromGitHub {
     owner = "PurpleI2P";
     repo = pname;
     rev = version;
-    sha256 = "02nyk76q2ag0495ph62i0jij27nxpy6qvryjp25wah8f69k7bgfs";
+    sha256 = "1yl5h7mls50vkg7x5510mljmgsm02arqhcanwkrqw4ilwvcp1mgz";
   };
 
-  buildInputs = [ boost zlib openssl ];
+  buildInputs = [ boost165 zlib openssl ];
```

Comment:
Thank you! Please override versions in `all-packages.nix`, like this:
https://github.com/NixOS/nixpkgs/blob/836951fc14e56c62cf5706d20ede240223db8f83/pkgs/top-level/all-packages.nix#L790-L792

Reply:
Oops. Should I override derivations which have a version in the name like this(`boost*`, `python*Packages`, ...) for every package I change? Is it a common approach? Asking because I have other PR where it is probably could be a good thing :)

Reply:
This is mostly useful for tracking which derivations depend on older versions (that will be deleted in the future once no package depends on them) and trying to switch them to newer versions of their dependencies en masse. This does not quite apply to python (python3 can be seen as less an upgrade and more a different thing, like qt4 and qt5, gtk2 and gtk3, gnupg1 and gnupg, grub and grub2), so IMHO both approaches are fine, and AFAIK referring to python3 in Python 3 derivations is more prevalent.